subsys/bluetooth/controller/ll_sw/isoal.c

```diff
@@ -1276,6 +1492,31 @@ static isoal_status_t isoal_tx_unframed_produce(struct isoal_source *source,
 		pp->payload_number = MAX(pp->payload_number,
 			(tx_sdu->target_event * session->burst_number));
 
+		/* Get actual event for this payload number */
+		actual_event = pp->payload_number / session->burst_number;
+
+		/* Get group reference point for this PDU based on the actual
+		 * event being set. This might introduce some errors as the
+		 * group refernce point for future events could drift. However
+		 * as the time offset calculation requires an absolute value,
+		 * this seems to be the best candidate.
+		 */
```

Comment:
Agreed, the changes that I made have caused the scope of this PR to creep, so I would like to propose that I will return it to the original scope with the comment we discussed.

There is a potential omission in considering the possibility of a wrap around in the framed TX, but this is in the code that is already upstream. So I will create a separate issue for that so that it can be  dealt with it in a separate PR.

Would that be acceptable?

Reply:
I'd like get input from other reviewers on this. I have a preference to fix this in this issue, as introducing new bugs in a `fix` commit just doesn't seem right :D

Reply:
I'm not certain that we are exactly on the same page here. :-) But yes lets gather the feedback from the other reviewers as well and then see how I can best address it.

Reply:
> I'd like get input from other reviewers on this. I have a preference to fix this in this issue, as introducing new bugs in a `fix` commit just doesn't seem right :D

I don't see this as "introducing new bugs". The comment in the code is perhaps unclear, but what is actually going on is that the code predicts the best possible group reference point by multiplying with the ISO interval, knowing that this is going to be slightly off due to drift. This is however by design, and the reference point is "corrected" along the way.